### PR TITLE
Right click a feature on the model to center the code pane on it #7447

### PR DESCRIPTION
**Pull request by rajgandhi1**
image

### CODE REVIEW

**Comment by rajgandhi1** [comment]
@nicboone8

**Comment by rajgandhi1** [comment]
@nicboone8 is the implementation correct? Do I need to modify something?

**Comment by rajgandhi1** [comment]
Hey @pierremtb 
Sure let me check the bot suggestions, sorry I didn't check them out before, and make the changes. Will also implement the tests.

**Comment by rajgandhi1** [comment]
@pierremtb I made the changes

File: e2e/playwright/view-kcl-source-code.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { getUtils } from '@e2e/playwright/test-utils'
+import { expect, test } from '@e2e/playwright/zoo-test'
+
+test.describe('View KCL source code functionality', () => {
+  test.setTimeout(90_000)
+
+  test('@web "View KCL source code" handles edge cases', async ({
```
Comment (pierremtb) [inline e2e/playwright/view-kcl-source-code.spec.ts:7]: Hey! Thanks for adding this @rajgandhi1 

Two things here:
1. I'd want to find a way not to add a new test with absolute coordinates as we're trying to slowly replace them with other approaches that are more robust to UI changes. For instance here, maybe finding the size of the scene (not the window, the video element really) and dividing that by two would probably work?
2. The three cases are good to have, but it's really the case where we can click and that checks on the highlight that I think we're missing here?

Those are things I'd be happy to help on as well, just lacking time at the moment but will likely be able to help soon. The next step will be for me to recreate this PR from your branch so we can get the tests to green

**Review by pierremtb — CHANGES_REQUESTED** [review]
